crates/red_knot_python_semantic/src/symbol.rs

```diff
@@ -501,15 +502,15 @@ fn symbol_from_bindings_impl<'db>(
             visibility_constraint,
             constraints: _,
         }) if binding.map_or(true, is_non_exported) => {
-            visibility_constraints.evaluate(db, *visibility_constraint)
+            visibility_constraints.evaluate(db, constraints, *visibility_constraint)
         }
         _ => Truthiness::AlwaysFalse,
     };
 
     let mut types = bindings_with_constraints.filter_map(
         |BindingWithConstraints {
              binding,
-             constraints,
+             constraints: binding_constraints,
```

Comment:
would `narrowing_constraints` also be a clearer name here, given that visibility constraints also apply to a binding?)

(and if we used that name consistently, the unpack would be simpler too)